plugins/mediatek-scaler/fu-mediatek-scaler-device.c

```diff
@@ -363,79 +348,177 @@ fu_mediatek_scaler_device_get_firmware_version(FuDevice *device, GError **error)
 	if (!fu_memread_uint32_safe(st_res->data,
 				    st_res->len,
 				    2,
-				    &tmpversion,
+				    &version_raw,
 				    G_LITTLE_ENDIAN,
 				    error))
 		return FALSE;
-	fw_version = mediatek_scaler_device_version_to_string(tmpversion);
-	fu_device_set_version(device, fw_version);
+	fu_device_set_version_raw(FU_DEVICE(self), version_raw);
 	return TRUE;
 }
 
 static gboolean
-fu_mediatek_scaler_device_probe(FuDevice *device, GError **error)
+fu_mediatek_scaler_device_open(FuDevice *device, GError **error)
 {
 	FuMediatekScalerDevice *self = FU_MEDIATEK_SCALER_DEVICE(device);
-	g_autoptr(FuUdevDevice) udev_parent = NULL;
-	g_autoptr(FuDeviceLocker) locker = NULL;
-	g_autofree gchar *vendor_id = NULL;
-	g_autofree gchar *vid = NULL;
-	g_autofree gchar *pid = NULL;
-	g_autofree gchar *hw_ver = NULL;
 
-	/* FuUdevDevice->probe */
-	if (!FU_DEVICE_CLASS(fu_mediatek_scaler_device_parent_class)->probe(device, error))
+	/* proxy */
+	if (!fu_device_open(FU_DEVICE(self->i2c_dev), error))
 		return FALSE;
 
-	/* determine the i2c_dev for dp aux dev */
-	if (!fu_mediatek_scaler_device_use_aux_dev(device, error))
+	/* set the target address */
+	if (!fu_mediatek_scaler_ensure_device_address(self,
+						      FU_DDC_I2C_ADDR_DISPLAY_DEVICE >> 1,
+						      error))
 		return FALSE;
 
-	/* open device */
-	locker = fu_device_locker_new(self->i2c_dev, error);
-	if (locker == NULL)
+	/* prioritize DDC/CI commands in display controller */
```

Comment:
We can move `set_priority()` to `_setup()` right after a mediatek display is confirmed in connected.

Reply:
I don't understand this bit; surely you want the priority to be set on every ->open() and unset on every ->close()? if you do it in ->setup() it'll only be done once at enumerate time.

Reply:
Device put in a state addressing ddc commands at priority, resulted in freeze/button no function etc. So we should set it for mediatek controllers only that is known at setup.

Reply:
Ohh I see, good point. Let me ponder a minute.

Reply:
@CragW does VCP_CODE_CONTROLLER_TYPE require us to have called fu_mediatek_scaler_ensure_device_address() first?

Reply:
Yes, set the slave address afterwards pwrite for device communication.

Reply:
Okay, so we need to do something about ->close too -- we're setting the FU_DDCCI_PRIORITY back to normal on ->close. Does my fixup make sense?

Reply:
That makes sensible! I've added a small change for unsupported device in `->close()`.

Reply:
Teamwork FTW! Thanks.